### PR TITLE
fix!: drop support for Python 2.7

### LINKED ISSUE #777: Drop Python 2.7 support
This would be semver major.

We should verify that clients which cannot move already pin `google.auth < 2.0.0dev` directly, or transitively.

Extra cleanups:

.- ~~Drop `mock` wrapper (use `from unittest import mock` directly).~~ We cannot drop `mock` because Python 3.6 / 3.7 do not have `unittest.mock.AsyncMock` in the stdlib.
- [x] Drop `six`.
- [x] Drop `u"` prefixes for text strings.

### PR DESCRIPTION
**Pull request by tseaver**
- Drop use of `six` wrapper library.
- Use editable installs for tests.
- Drop use of `u"` prefixes.

Release-As: 2.0.0b1
    
Closes #777.

### CODE REVIEW

**Comment by parthea** [comment]
I'm planning to use suffix `-rc1` when major version >= 1 . Could we do the same here or are you also planning to also use `rcX` after `bX`? In other words, would the versioning be:

```
2.0.0rc1
2.0.0rc2
```
or
```
2.0.0b1
2.0.0b2
followed by
2.0.0rc1
2.0.0rc2
```

**Comment by tseaver** [comment]
@parthea  I wasn't planning to use `rc` for the intial Py3-only release -- we need to have some shakeout time for the dependent libraries, which makes this more of a beta than a release candidate.

**Comment by tseaver** [comment]
I'm having difficulty reproducing the [Kokoro systest failures](https://source.cloud.google.com/results/invocations/980186ee-982b-4d86-87a2-b1f39d0f2807/targets/cloud-devrel%2Fclient-libraries%2Fpython%2Fgoogleapis%2Fgoogle-auth-library-python%2Fpresubmit%2Fpresubmit/log).  The failing sessions:

- `default_explicit_authorized_user`
- `default_explicit_authorized_user_explicit_project`
- `default_explicit_authorized_user_async`
- `default_explicit_authorized_user_explicit_project_async`

all pass for me when run locally:

```bash
$ nox -f system_tests/noxfile.py -e default_explicit_authorized_user default_explicit_authorized_user_explicit_project default_explicit_authorized_user_async default_explicit_authorized_user_explicit_project_async
...
nox > Ran multiple sessions:
nox > * default_explicit_authorized_user-3.7: success
nox > * default_explicit_authorized_user_explicit_project-3.7: success
nox > * default_explicit_authorized_user_async-3.7: success
nox > * default_explicit_authorized_user_explicit_project_async-3.7: success
```

@busunkim96, @arithmetic1728, @crwilcox can any of you see what is happening?

**Comment by arithmetic1728** [comment]
It seems the refresh token used in kokoro system test is expired.
```
google.auth.exceptions.RefreshError: ('invalid_grant: Token has been expired or revoked.', {'error': 'invalid_grant', 'error_description': 'Token has been expired or revoked.'})
```

File: google/auth/version.py [hunk 1]
```diff
@@ -12,4 +12,4 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-__version__ = "1.34.0"
+__version__ = "2.0.0b1"
```
Comment (busunkim96) [inline google/auth/version.py:15]: I would prefer the version be bumped in a separate release PR  (through `Release-As: 2.0.0b1`)

**Review by busunkim96 — APPROVED** [review]
Please let @arithmetic1728 also take a look.

**Review by arithmetic1728 — APPROVED** [review]
LGTM

File: google/auth/version.py (hunk 1 above)
Comment (tseaver) [reply google/auth/version.py]: 35f4546

**Pull request merged**